sql/sql_class.cc

```diff
@@ -6197,6 +6197,11 @@ int THD::decide_logging_format(TABLE_LIST *tables)
   DBUG_RETURN(0);
 }
 
+sql_mode_t sql_mode_for_dates(THD *thd)
+{
+	return thd->variables.sql_mode &
+		(MODE_NO_ZERO_DATE | MODE_NO_ZERO_IN_DATE | MODE_INVALID_DATES);
+}
```

Comment:
This function is missing a comment. What is this function needed for?
I see that `innobase_get_field_def_value()` is calling it. Why should the default value of the column depend on the caller’s state? I think that the exact default value must be persisted in `SYS_COLUMNS_ADDED`, and it must be returned as is, without referring to `current_thd`.